var/spack/repos/builtin/packages/bazel/bazelconfiguration-0.3.patch

```diff
@@ -0,0 +1,16 @@
+--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java
++++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java
+@@ -150,6 +150,13 @@
+       builder.put("PATH", null);
+       builder.put("LD_LIBRARY_PATH", null);
+     }
++
++    Map<String, String> env = System.getenv();
++    for (String envName : env.keySet()) {
++      if (envName.startsWith("SPACK_")) {
++        builder.put(envName, env.get(envName));
++      }
++    }
```

Comment:
Previously, we enumerated through every Spack environment variable and passed it to the build if it was set. This was extremely error prone, as the exact env vars Spack sets in `build_environment.py` and uses in the compiler wrappers changes frequently. About half of the env vars we were setting no longer exist, and the other half weren't being passed.

With this newer patch, bazel will pass all `SPACK_*` env vars to the build environment. I haven't written Java in a _loooong_ time, but I think I did things right.

Reply:
Great improvement!